azurelinuxagent/ga/logcollector.py

```diff
@@ -320,6 +320,8 @@ def _get_final_list_for_archive(self, priority_file_queue):
                     if truncated_file_path:
                         _LOGGER.info("Adding truncated file %s, size %s b", truncated_file_path, file_size)
                         final_files_to_collect.append(truncated_file_path)
+                    else:
```

Comment:
While updating the unit tests to check uncompressed_file_size I found a bug where we are adding the file size of files we will not collect (binaries) to the total uncompressed size. Fixing this here

Reply:
Good catch.

The code looks so weird, though, especially without an explanation, and with that "if truncated_file_path". No need to clean that up, but doing "total_uncompressed_size += file_size" only when "final_files_to_collect.append()" may be a little more readable.